packages/core/src/compiler/Style.ts

```diff
@@ -2690,20 +2633,25 @@ const initShape = (
   const res = findExprSafe(tr, path); // This is safe (as used in GenOptProblem) since we only initialize shapes with paths from the translation
 
   if (res.tag === "FGPI") {
-    const [stype, props] = res.contents as [string, GPIProps<VarAD>];
-    const def: ShapeDef = findDef(stype);
-    const gpiTemplate: [string, [PropType, Sampler]][] = Object.entries(
-      def.properties
-    );
-
-    const instantiatedGPIProps: GPIProps<VarAD> = gpiTemplate.reduce(
-      (
-        newGPI: GPIProps<VarAD>,
-        propTemplate: [string, [PropType, Sampler]]
-      ): GPIProps<VarAD> =>
-        initProperty(stype, newGPI, propTemplate, getCanvas(tr)),
-      clone(props)
-    ); // NOTE: `initProperty` mutates its input, so the `props` from the translation is cloned here, so the one in the translation itself isn't mutated
+    const [stype, props] = res.contents;
+    const instantiatedGPIProps: GPIProps<VarAD> = {
+      ...Object.fromEntries(
+        Object.entries(
+          shapedefs[stype].sampler(getCanvas(tr))
+        ).map(([propName, contents]) => [
+          propName,
+          { tag: isPending(stype, propName) ? "Pending" : "Done", contents },
+        ])
+      ),
+      ...Object.fromEntries(
```

Comment:
My understanding is the entries above will contain all the properties in a shape, the then they will be overwritten by the entries below, if they are set by Style? Either way, it would be good to have a comment describing the intended behavior. A bit hard to decipher what this is doing.

Reply:
OK! I'll add a comment.